.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,53 @@
+name: E2E
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+
+env:
+  KUBEBUILDER_CONTROLPLANE_STOP_TIMEOUT: 120s
+
+jobs:
+  e2e:
+    name: E2E
+    runs-on: ubuntu-24.04
+    steps:
+    - name: Check out
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version-file: 'go.mod'
+    - name: Install ginkgo
+      run: go install github.com/onsi/ginkgo/v2/ginkgo@v2.1.4 # This is aligned with the version in go.mod
+    - name: Install kind
+      run: curl -Lo kind https://kind.sigs.k8s.io/dl/${{ env.KIND_VERSION }}/kind-linux-amd64 && chmod +x kind && sudo mv kind /usr/local/bin/
+      env:
+        KIND_VERSION: v0.27.0
+    - name: Install krew
+      run: |
+        set -x; cd "$(mktemp -d)" &&
+        OS="$(uname | tr '[:upper:]' '[:lower:]')" &&
+        ARCH="$(uname -m | sed -e 's/x86_64/amd64/' -e 's/\(arm\)\(64\)\?.*/\1\2/' -e 's/aarch64$/arm64/')" &&
+        KREW="krew-${OS}_${ARCH}" &&
+        curl -fsSLO "https://github.com/kubernetes-sigs/krew/releases/download/${KREW_VERSION}/${KREW}.tar.gz" &&
+        tar zxvf "${KREW}.tar.gz" &&
+        ./"${KREW}" install krew
+        echo "$HOME/.krew/bin" >> $GITHUB_PATH
+      env:
+        KREW_VERSION: v0.4.5
```

Comment:
I think these tools should be installed as dependencies of `make test-e2e` because they are also necessary when running e2e tests outside of CI.